ingestors/kafka/src/main/scala/hydra/kafka/algebras/KafkaClientAlgebra.scala

```diff
@@ -205,7 +212,7 @@ object KafkaClientAlgebra {
       }
 
       override def withProducerRecordSizeLimit(sizeLimitBytes: Long): F[KafkaClientAlgebra[F]] =
-        getLiveInstance[F](bootstrapServers)(queue, schemaRegistryClient, keySerializer, valSerializer, sizeLimitBytes.some, publishTimeoutDuration)
+        getLiveInstance[F](bootstrapServers)(queue, schemaRegistryClient, keySerializer, valSerializer, sizeLimitBytes.some, publishTimeoutDuration, publishMaxBlockMs)
```

Comment:
Just curious, what's the difference between publishTimeoutDuration and publishMaxBlockMs?

Reply:
The timeout is the max amount of time that it will wait before returning a timeout, the maxBlockMs is directly setting `max.block.ms` on the Kafka producer.
From the docs for `max.block.ms`:

>The configuration controls how long {@link KafkaProducer#send()} and {@link KafkaProducer#partitionsFor} will block.These methods can be blocked either because the buffer is full or metadata unavailable.Blocking in the user-supplied serializers or partitioner will not be counted against this timeout.